### PR TITLE
Issue 2 item index

### PR DESCRIPTION
**Pull request by CjMoore**
Adds Items feature test for index, items controller, items index view, item model and model tests.

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class ItemsController < ApplicationController
+
+  def index
```
Comment (cmacaulay) [inline app/controllers/items_controller.rb]: 👍

File: app/models/item.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Item < ApplicationRecord
+
+  validates :title, :description, :price, presence: true
+  validates :title, :description, uniqueness: true
```
Comment (cmacaulay) [inline app/models/item.rb]: Looks good! @CjMoore are you thinking we add image validations at some point? When I add categories, should I add a validation for that as well for item?

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -2,6 +2,9 @@
 <html>
   <head>
     <title>WeasleysWizardWheezes</title>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
```
Comment (cmacaulay) [inline app/views/layouts/application.html.erb]: Can't wait to see what these are!

File: db/migrate/20170221225907_create_items.rb [hunk 4]
```diff
@@ -3,7 +3,7 @@ def change
     create_table :items do |t|
       t.string :title
       t.string :description
-      t.decimal :price, :precision => 2
+      t.float :price
```
Comment (cmacaulay) [inline db/migrate/20170221225907_create_items.rb]: @CjMoore are we keeping this a float for now? Still interested in potentially using wizard money, but then we would have to re-assess trying to do the stripe or paypal extension @meyerhoferc @valerietrudell

File: spec/features/user_can_view_all_items_spec.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+describe "When user visits root" do
```
[the remark this answers is not in the document]
Comment (cmacaulay) [reply spec/features/user_can_view_all_items_spec.rb:3]: rude

File: spec/models/item_spec.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+    context "is invalid" do
+      it "without a title" do
+        item = Item.new( description: "a rocket from the Explosive Enterprises line by Weasleys' Wizard Wheezes", price: 10.99 )
+
+        expect(item).to be_invalid
+      end
+
+      it "without description" do
+        item = Item.new(title: "Bang Bang Boggart Banger",  price: 10.99 )
```
Comment (cmacaulay) [inline spec/models/item_spec.rb]: Amazing item.

**Review by cmacaulay — COMMENTED** [review]
Looks good @CjMoore! Looks like hound is houndin' you, how do we want to approach those changes?

File: db/migrate/20170221225907_create_items.rb (hunk 4 above)
Comment (CjMoore) [reply db/migrate/20170221225907_create_items.rb]: Yeah, I couldn't get decimal to work properly. We can do a new migration if we figure out a better strategy for storing price down the line.

**Comment by meyerhoferc** [comment]
this looks good. wondering what we want to do about hound violations? @CjMoore @cmacaulay @valerietrudell

**Pull request merged**